proposed/2020/PackageVulnerability/FlagVulnerablePackages.md

```diff
@@ -0,0 +1,99 @@
+
+## Goals
+
+This is a security feature aiming to fill a gap in the .NET package management ecosystem known as *package vulnerability auditing*.
+
+NuGet should be able to flag vulnerable packages used in projects/solutions.
+
+The ultimate goal is to *reduce exposure to known vulnerabilities* in NuGet packages, and *increase adoption rate of fixes* for those known vulnerabilities.
+
+## Non-Goals
+
+* At this point, NuGet.org won't have a way for users to input vulnerability data into the system. All vulnerability data will be sourced from external systems. (no self-reporting)
+* This feature is an enhancement to the NuGet v3 protocol. We won't surface this in earlier versions of the NuGet protocol.
+* `NuGet.Server` is out-of-scope for this feature at this point.
+
+## Scope
+
+There are multiple scenarios for flagging and reporting vulnerabilities. This feature will be implemented in several stages which will enable a subset of those as discussed below.
+
+### Stage 1 - MVP
+
+For the [MVP](https://en.wikipedia.org/wiki/Minimum_viable_product) of this feature, we'll focus on:
+
+* Enabling ***consumers** to be informed about known vulnerabilities before installation* of a package.
+* Enabling ***authors** to be informed about known vulnerabilities in their packages*.
+* Leveraging **existing, curated, and publicly available known vulnerabilities** affecting NuGet packages.
+
+This should help prevent *new* installs of packages that are known to be vulnerable, and trigger authors into providing a fix through a new package version on nuget.org.
+
+NuGet will not support self-reporting of vulnerabilities at this time and instead rely on an existing data provider of curated vulnerability information. For the MVP of this feature, all vulnerability data will be sourced from external systems. The initial focus is on leveraging [GitHub's Security Workflow](https://github.com/features/security) and integration with [GitHub's GraphQL API](https://developer.github.com/v4/). This allows us to build an initial end-to-end experience without having to deal with the complexities of curating vulnerability reports.
+
+#### ✔️ In scope (Stage 1)
+
+* For CLI:
+  * A CLI command to **list vulnerabilities** in a project/solution project graph
+
+* For Visual Studio:
+  * **Display vulnerability information** in the package graph on `Installed` tab of Package Manager Dialog (PMUI)
+  * **Display vulnerability information** for *transitive packages* in PMUI (additional UX complexity) - note that this is dependent on transitive packages being displayed in general, which is to be specced separately
+
+* For nuget.org:
+  * **Automatically ingest and refresh NuGet package vulnerability data from GitHub**
+  * **Surface package vulnerability information on-demand**:
+    * as part of package metadata in `v3` registration blobs (`JSON`)
+    * by showing vulnerability information to package owners (only!) on package details page and version history
```

Comment:
Are we still planning to show this info to authors only? /cc: @JonDouglas

Reply:
Good catch, @anangaur --I've removed this now, as well as the email bullet.